zipline/pipeline/loaders/blaze/core.py

```diff
@@ -729,25 +729,36 @@ def __new__(cls,
             expr,
             deltas,
             checkpoints,
-            odo_kwargs or {},
+            frozenset((odo_kwargs or {}).items()),
         )
 
     def __repr__(self):
         # If the expressions have _resources() then the repr will
         # drive computation so we take the str here.
-        cls = type(self)
-        return super(ExprData, cls).__repr__(cls(
+        return repr(_expr_data_base(
```

Comment:
Why this change over the `super`-based implementation?

Reply:
I actually want to call the super class's constructor and then repr that. I don't want to do the `dict` to `frozenset` conversion. This seemed like the most straightforward way to do that. I am not that worried about some complex hierarchy below this type.